### PR TITLE
fix: cluster ID used in demo application

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

cluster ID is used in the demo in multicluster mode, which breaks the demo. This is a continuation of #3854 .

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [x] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

No 

1. Is this a breaking change?

No

### CODE REVIEW

**Review by draychev — APPROVED** [review]

File: pkg/injector/envoy_config_test.go [hunk 1]
```diff
@@ -29,7 +29,6 @@ var _ = Describe("Test functions creating Envoy bootstrap configuration", func()
 	const (
 		containerName = "-container-name-"
 		envoyImage    = "-envoy-image-"
-		clusterID     = "-cluster-id-"
```
Comment (draychev) [inline pkg/injector/envoy_config_test.go:32]: We need to keep this here - this is used in the Envoy config (it was here before Multicluster)

File: pkg/injector/envoy_container.go [hunk 2]
```diff
@@ -55,7 +52,6 @@ func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator
 		Args: []string{
 			"--log-level", cfg.GetEnvoyLogLevel(),
 			"--config-path", strings.Join([]string{envoyProxyConfigPath, envoyBootstrapConfigFile}, "/"),
-			"--service-cluster", clusterID,
```
Comment (draychev) [inline pkg/injector/envoy_container.go:58]: We need to keep this here as well. This predates Multicluster - it is needed by Envoy.

**Review by draychev — COMMENTED** [review]
@allenlsy would you be so kind to revert the changes to pkg/injector/envoy_container.go -- these predate Multicluster; Envoy needs a `service-cluster`

File: pkg/injector/envoy_config_test.go (hunk 1 above)
Comment (allenlsy) [reply pkg/injector/envoy_config_test.go:32]: I'm looking for what breaks the CI. Thanks.

**Review by snehachhabria — APPROVED** [review]

File: pkg/injector/envoy_container.go [hunk 3]
```diff
@@ -35,8 +34,6 @@ func getPlatformSpecificSpecComponents(cfg configurator.Configurator, podOS stri
 }
 
 func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes, podOS string) corev1.Container {
-	// cluster ID will be used as an identifier to the tracing sink
```
Comment (shashankram) [inline pkg/injector/envoy_container.go:38]: This should not be removed.

File: pkg/injector/envoy_container.go (hunk 2 above)
Comment (shashankram) [reply pkg/injector/envoy_container.go:58]: +1

**Review by shashankram — CHANGES_REQUESTED** [review]

File: pkg/injector/envoy_container.go (hunk 2 above)
Comment (snehachhabria) [reply pkg/injector/envoy_container.go:58]: +1

File: pkg/injector/envoy_container.go [hunk 4]
```diff
@@ -35,9 +35,9 @@ func getPlatformSpecificSpecComponents(cfg configurator.Configurator, podOS stri
 }
 
 func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes, podOS string) corev1.Container {
+	securityContext, containerImage := getPlatformSpecificSpecComponents(cfg, podOS)
```
Comment (shashankram) [inline pkg/injector/envoy_container.go:38]: Please drop this change.

**Review by shashankram — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**